icu4c/source/common/rbbitblb.cpp

```diff
@@ -1378,18 +1392,29 @@ void RBBITableBuilder::exportTable(void *where) {
     for (state=0; state<table->fNumStates; state++) {
         RBBIStateDescriptor *sd = (RBBIStateDescriptor *)fDStates->elementAt(state);
         RBBIStateTableRow   *row = (RBBIStateTableRow *)(table->fTableData + state*table->fRowLen);
-        U_ASSERT (-32768 < sd->fAccepting && sd->fAccepting <= 32767);
-        U_ASSERT (-32768 < sd->fLookAhead && sd->fLookAhead <= 32767);
-        row->fAccepting = (int16_t)sd->fAccepting;
-        row->fLookAhead = (int16_t)sd->fLookAhead;
-        row->fTagIdx    = (int16_t)sd->fTagsIdx;
-        for (col=0; col<catCount; col++) {
-            row->fNextState[col] = (uint16_t)sd->fDtran->elementAti(col);
+        if (use8BitsForTable()) {
+            U_ASSERT (-128 < sd->fAccepting && sd->fAccepting < kStateThresholdFor8BitsTable);
+            U_ASSERT (-128 < sd->fLookAhead && sd->fLookAhead < kStateThresholdFor8BitsTable);
+            U_ASSERT (-128 < sd->fTagsIdx && sd->fTagsIdx <= kStateThresholdFor8BitsTable);
+            row->s8.fAccepting = (int8_t)sd->fAccepting;
+            row->s8.fLookAhead = (int8_t)sd->fLookAhead;
+            row->s8.fTagIdx    = (int8_t)sd->fTagsIdx;
+            for (col=0; col<catCount; col++) {
+                row->s8.fNextState[col] = (int8_t)sd->fDtran->elementAti(col);
```

Comment:
similar to question in runtime code: why signed? why not uint8_t?

Reply:
ping -- especially now that r8.fNextState[] is really a uint8_t

Reply:
I have code in another branch that changes all of the state table row fields to unsigned. I would like to do that in a separate PR, because this one has become too bit and unwieldy - meaning let this stay as it is in this PR.